File: kangtaku/largest_number_at_least_twice_of_others.java
class Solution {
    public int dominantIndex(int[] nums) {
        int max = -1, mIdx = -1;
        for (int i = 0; i < nums.length; i++) {
            if (max < nums[i]) {
                max = nums[i];
                mIdx = i;
            }
        }
        for (int i = 0; i < nums.length; i++) {
            if (i != mIdx && max < nums[i] * 2) {
                return -1;
            }
        }
        return mIdx;
    }
}

File: iamminji/leetcode/easy/largest_number_at_least_twice_of_others.py
# 747. Largest Number At Least Twice of Others
# https://leetcode.com/problems/largest-number-at-least-twice-of-others/description/


class Solution:
    def dominantIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        max_num = max(nums)
        res = -1
        for i, num in enumerate(nums):
            if num == max_num:
                res = i
            elif num * 2 > max_num:
                res = -1
                break
        return res
